storybookjs/storybook
# Fixes component imports from dist

Issue: N/A

## What I did

There was a bug with the webpack plugin `TsconfigPathsPlugin` where Storybook in Angular could not import angular components when they are built.

This was my test scenario
```typescript
// Starting from a projects/index.stories.ts file

// Fails (path: ./dist/lib-a)
import { LibAModule } from 'lib-a';

// Works (path: /projects/lib-a/src/lib/lib-a.module.ts)
import { LibAModule } from '../lib-a/src/lib/lib-a.module';
```

After checking the angular values here: https://github.com/angular/angular-cli/blob/9f114aee1e009c3580784dd3bb7299bdf4a5918c/packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/server.ts#L21 

I added

```typescript
      new TsconfigPathsPlugin({
        configFile: cliWebpackConfigOptions.buildOptions.tsConfig,
        // After ng build my-lib the default value of 'main' in the package.json is 'umd'
        // This causes that you cannot import components directly from dist
        // https://github.com/angular/angular-cli/blob/9f114aee1e009c3580784dd3bb7299bdf4a5918c/packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/browser.ts#L68
        mainFields: [
          ...(cliWebpackConfigOptions.supportES2015 ? ['es2015'] : []),
          'browser',
          'module',
          'main',
        ],
```

## How to test

- Checkout https://github.com/kroeder/storybook-tsconfig-paths-webpack-issue-repro
- Follow the readme up until 4. to see the issue
- `link` this change into the app and start storybook again, should work
